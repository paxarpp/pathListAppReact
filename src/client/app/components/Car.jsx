import React from 'react';
import PropTypes from 'prop-types';

import Button from '../components/Button';

const Car = ({ car, handler, selectedCar, deleteCarHandler }) => {
  return (
    <div onClick={handler} className={selectedCar === car.name ? 'selected' : ''}>
      {car.name}
      <Button handler={deleteCarHandler(car.name)} styleButton="delit">
        {String.fromCharCode(10006)}
      </Button>
    </div>
  );
};
Car.propTypes = {
  car: PropTypes.object,
  selectedCar: PropTypes.string,
  handler: PropTypes.func,
  deleteCarHandler: PropTypes.func
};

export default Car;
