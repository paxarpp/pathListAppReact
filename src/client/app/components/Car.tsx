import React from 'react';
import styled, { css } from 'styled-components';
import { ICar } from './interfaces';
import Icon from './Icon';

interface IProp {
  car: ICar;
  selectedCar: string;
  handler: (h: any) => any;
  deleteCarHandler: (name: string) => any;
};

interface WrapProp {
  isSelectedCar: boolean;
};

const Car = ({ car, handler, selectedCar, deleteCarHandler }: IProp) => (
  <Wrap onClick={handler} isSelectedCar={selectedCar === car.name}>
    {car.name}
    <Icon name="Delete" color="red" onClick={deleteCarHandler(car.name)} />
  </Wrap>
);

const selected = css`
  background-color: #fff;
  width: 100%;
  border-radius: 20px 0 0 20px;
  position: relative;
  transition: background-color 0.2s;
  :after {
    content: '';
    position: absolute;
    top: -10px;
    right: 0;
    border: 5px solid transparent;
    border-right: 5px solid #fff;
    border-bottom: 5px solid #fff;
  }
  :before {
    content: '';
    border: 5px solid transparent;
    border-top: 5px solid #fff;
    border-right: 5px solid #fff;
    position: absolute;
    bottom: -10px;
    right: 0;
  }
`;

const Wrap = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  align-items: center;
  padding-left: 1rem;
  ${(props: WrapProp) => props.isSelectedCar && selected};
`;

export default Car;
