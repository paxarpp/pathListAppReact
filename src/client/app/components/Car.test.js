import React from 'react';
import renderer from 'react-test-renderer';
import Car from './Car';

test('Component no the class', () => {
  const car = { name: 'test' };
  const selectedCar = '';
  const component = renderer.create(
    <Car car={car} selectedCar={selectedCar} deleteCarHandler={() => {}} handler={() => {}} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Component has the class when select', () => {
  const car = { name: 'test' };
  const selectedCar = 'test';
  const component = renderer.create(
    <Car car={car} selectedCar={selectedCar} deleteCarHandler={() => {}} handler={() => {}} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
