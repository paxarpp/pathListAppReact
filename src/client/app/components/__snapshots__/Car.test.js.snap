// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component has the class when select 1`] = `
<div
  className="selected"
  onClick={[Function]}
>
  test
  <a
    className="delit"
  >
    ✖
  </a>
</div>
`;

exports[`Component no the class 1`] = `
<div
  className=""
  onClick={[Function]}
>
  test
  <a
    className="delit"
  >
    ✖
  </a>
</div>
`;
